hir2/src/ir/context.rs

```diff
@@ -276,6 +276,35 @@ impl Context {
         }
     }
 
+    /// Changes the destination of the `branch_inst` instruction to `new_dest` if the current
+    /// destination is `old_dest`.
+    ///
+    /// NOTE: Panics if `branch_inst` is not a branch instruction.
+    pub fn change_branch_destination(
```

Comment:
I don't think we should attach this to `Context`. I would make this a new trait method of `BranchOpInterface` instead.

Additionally, we need to ensure that `old_dest` and `new_dest` have the same number of successor operands, otherwise the resulting IR will be invalid.

Reply:
Yep, it's much nicer. Thanks! Done.